### PR TITLE
Avoid double slash when join url and path (#5503)

### LINKED ISSUE #5503: ES URL with Trailing Slash Causes Exception
Adding an slash to the end of the configured ES URL causes an exception in ES when the template is installed at startup. It looks like the Beat uses a URL with a double slash in it like `http://localhost:9200//_template/auditbeat-6.0.0-rc`. This double slash can be avoided during the construction of the URL within Beats.

For confirmed bugs, please report:
- Version: 6.0.0-rc2 tested with ES 5.4.1
- Steps to Reproduce: Append a slash to the ES URL.

```
output.elasticsearch:
  hosts: ["10.100.6.80:9200/"]
```

`sudo ./beat setup --template -e -d "*" --strict.perms=false`

From Beats:

```
2017/11/02 16:46:45.394290 beat.go:625: CRIT Exiting: Error loading Elasticsearch template: could not load template: couldn't load template: couldn't load json. Error: 500 Internal Server Error. Response body: {"error":{"root_cause":[{"type":"string_index_out_of_bounds_exception","reason":"String index out of range: 0"}],"type":"string_index_out_of_bounds_exception","reason":"String index out of range: 0"},"status":500}
Exiting: Error loading Elasticsearch template: could not load template: couldn't load template: couldn't load json. Error: 500 Internal Server Error. Response body: {"error":{"root_cause":[{"type":"string_index_out_of_bounds_exception","reason":"String index out of range: 0"}],"type":"string_index_out_of_bounds_exception","reason":"String index out of range: 0"},"status":500}
```

From ES:

```
[2017-11-02T12:46:45,380][WARN ][r.suppressed             ] path: //_template/auditbeat-6.0.0-rc2, params: {index=, id=auditbeat-6.0.0-rc2, type=_template}
java.lang.StringIndexOutOfBoundsException: String index out of range: 0
	at java.lang.String.charAt(String.java:658) ~[?:1.8.0_111]
	at org.elasticsearch.cluster.metadata.MetaDataCreateIndexService.validateIndexOrAliasName(MetaDataCreateIndexService.java:167) ~[elasticsearch-5.4.1.jar:5.4.1]
	at org.elasticsearch.cluster.metadata.MetaDataCreateIndexService.validateIndexName(MetaDataCreateIndexService.java:142) ~[elasticsearch-5.4.1.jar:5.4.1]
	at org.elasticsearch.cluster.metadata.MetaDataCreateIndexService.validate(MetaDataCreateIndexService.java:495) ~[elasticsearch-5.4.1.jar:5.4.1]
	at org.elasticsearch.cluster.metadata.MetaDataCreateIndexService.access$000(MetaDataCreateIndexService.java:106) ~[elasticsearch-5.4.1.jar:5.4.1]
	at org.elasticsearch.cluster.metadata.MetaDataCreateIndexService$1.execute(MetaDataCreateIndexService.java:239) ~[elasticsearch-5.4.1.jar:5.4.1]
	at org.elasticsearch.cluster.ClusterStateUpdateTask.execute(ClusterStateUpdateTask.java:45) ~[elasticsearch-5.4.1.jar:5.4.1]
	at org.elasticsearch.cluster.service.ClusterService.executeTasks(ClusterService.java:633) ~[elasticsearch-5.4.1.jar:5.4.1]
	at org.elasticsearch.cluster.service.ClusterService.calculateTaskOutputs(ClusterService.java:611) ~[elasticsearch-5.4.1.jar:5.4.1]
	at org.elasticsearch.cluster.service.ClusterService.runTasks(ClusterService.java:570) [elasticsearch-5.4.1.jar:5.4.1]
	at org.elasticsearch.cluster.service.ClusterService$ClusterServiceTaskBatcher.run(ClusterService.java:262) [elasticsearch-5.4.1.jar:5.4.1]
	at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:150) [elasticsearch-5.4.1.jar:5.4.1]
	at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:188) [elasticsearch-5.4.1.jar:5.4.1]
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:569) [elasticsearch-5.4.1.jar:5.4.1]
	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:247) [elasticsearch-5.4.1.jar:5.4.1]
	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:210) [elasticsearch-5.4.1.jar:5.4.1]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_111]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_111]
```

### PR DESCRIPTION
**Pull request by liketic**
Closes #5503

### CODE REVIEW

File: libbeat/outputs/elasticsearch/client.go [hunk 1]
```diff
@@ -84,7 +84,7 @@ var (
 	errExpectedItemObject    = errors.New("expected item response object")
 	errExpectedStatusCode    = errors.New("expected item status code")
 	errUnexpectedEmptyObject = errors.New("empty object")
-	errExcpectedObjectEnd    = errors.New("expected end of object")
+	errExpectedObjectEnd     = errors.New("expected end of object")
```
Comment (ruflin) [inline libbeat/outputs/elasticsearch/client.go]: Thanks for the cleanup

**Comment by liketic** [comment]
@ruflin Thanks. I pushed 734ad81094a0f4053b57fe81164f955db8be9288

**Comment by ruflin** [comment]
@liketic It seems you have to run `make fmt` on this PR. I pushed 2 additional commits to fix the merge conflict in the CHANGELOG as I hoped to directly merge. You need to pull these changes or just overwrite them if you push again.

**Comment by liketic** [comment]
@ruflin Thanks for your kind help. I think everything is OK now. 👍

**Pull request merged**